packages/synapse-react-client/src/components/Authentication/AuthenticationMethodSelection.tsx

```diff
@@ -2,20 +2,40 @@ import SynapseClient from '@/synapse-client'
 import { OAuth2State, SynapseClientError } from '@/utils'
 import {
   LOGIN_METHOD_EMAIL,
+  LOGIN_METHOD_OAUTH2_ARCUS,
   LOGIN_METHOD_OAUTH2_GOOGLE,
   LOGIN_METHOD_OAUTH2_ORCID,
   OAUTH2_PROVIDERS,
 } from '@/utils/SynapseConstants'
 import { Box } from '@mui/material'
-import { MouseEvent } from 'react'
+import { MouseEvent, useMemo } from 'react'
 import LoginMethodButton from './LoginMethodButton'
 
+const CSRF_TOKEN_STORAGE_KEY = 'oauth2_csrf_token'
+
+function generateCsrfToken(): string {
+  const { crypto } = window
+  if (crypto?.getRandomValues) {
+    const bytes = new Uint8Array(32)
+    crypto.getRandomValues(bytes)
+    return Array.from(bytes, byte => byte.toString(16).padStart(2, '0')).join(
+      '',
+    )
+  }
+  if (crypto?.randomUUID) {
+    return crypto.randomUUID().replace(/-/g, '')
+  }
```

Comment:
Which is preferred? `getRandomValues` is more widely supported than `randomUUID`, so I think `randomUUID` would almost never be used here.